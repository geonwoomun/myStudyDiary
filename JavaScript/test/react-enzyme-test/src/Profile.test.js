import React from 'react';
import { mount } from 'enzyme';
import Profile from './Profile';

describe('<Profile/>', () => {
    it('matches snapshot', () => {
        const wrapper = mount(<Profile username="mun" name="문건우"/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders username and name', () => {
        const wrapper = mount(<Profile username="mun" name="문건우"/>);
        expect(wrapper.props().username).toBe('mun');
        expect(wrapper.props().name).toBe('문건우');

        const boldElement = wrapper.find('b');
        expect(boldElement.contains('mun')).toBe(true);
        const spanElement = wrapper.find('span');
        expect(spanElement.text()).toBe('(문건우)');
    });
});