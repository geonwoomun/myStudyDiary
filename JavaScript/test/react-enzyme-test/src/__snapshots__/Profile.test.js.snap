// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile/> matches snapshot 1`] = `
<Profile
  name="문건우"
  username="mun"
>
  <div>
    <b>
      mun
    </b>
    <span>
      (
      문건우
      )
    </span>
  </div>
</Profile>
`;
